### PR TITLE
feat: Add UDF reranker support

### PR DESCRIPTION
**Pull request by mrderyk**
## CONTEXT
We need to add support for the UDF reranker

## CHANGES
- Add UDF reranker types
- Add logic for converting reranker param to API-safe object
- Fix some misc type issues in docs and test files

### CODE REVIEW

File: docs/src/index.tsx [hunk 1]
```diff
@@ -73,20 +68,20 @@ const App = () => {
...
         },
-        reranker: "none",
+        reranker: { type: "none" }
       },
       generation: {
         maxUsedSearchResults: 5,
         responseLanguage: "eng",
         enableFactualConsistencyScore: true,
-        promptName: "vectara-experimental-summary-ext-2023-12-11-large",
+        generationPresetName: "vectara-experimental-summary-ext-2023-12-11-large"
```
Comment (mrderyk) [inline docs/src/index.tsx:79]: Type issue: This had the wrong property name.

File: docs/src/index.tsx [hunk 2]
```diff
@@ -35,11 +30,11 @@ const App = () => {
       language: "eng",
       chat: {
         store: true,
-        conversationId: conversationIdV1,
+        conversationId: conversationIdV1
       },
       debug: true,
       enableFactualConsistencyScore: true,
-      summaryPromptName: "vectara-experimental-summary-ext-2023-12-11-large",
+      summaryPromptName: "vectara-experimental-summary-ext-2023-12-11-large"
```
Comment (mrderyk) [inline docs/src/index.tsx:37]: Type issue: This had the wrong property name.

File: docs/src/index.tsx [hunk 3]
```diff
@@ -112,10 +107,7 @@ const App = () => {
 
     const queryStream = await streamQueryV2({
       streamQueryConfig,
-      onStreamEvent,
-      onError: (error) => {
-        console.error("Error", error);
-      },
+      onStreamEvent
```
Comment (mrderyk) [inline docs/src/index.tsx:110]: Type issue: We no longer support an `onError`. We should IMO and I'll add an issue for it.

File: src/apiV2/client.test.ts [hunk 4]
```diff
@@ -20,7 +20,7 @@ const streamQueryConfig: StreamQueryConfig = {
     maxUsedSearchResults: 5,
     responseLanguage: "eng",
     enableFactualConsistencyScore: true,
-    promptName: "vectara-experimental-summary-ext-2023-12-11-large"
+    generationPresetName: "vectara-experimental-summary-ext-2023-12-11-large"
```
Comment (mrderyk) [inline src/apiV2/client.test.ts:23]: Type issue: This had the wrong property name.

File: docs/src/index.tsx (hunk 3 above)
Comment (cjcenizal) [reply docs/src/index.tsx:110]: Errors are now emitted as part of the stream IIRC.

**Review by cjcenizal — APPROVED** [review]
Code LGTM. Didn't test locally.

**Comment by cjcenizal** [comment]
@mrderyk I forgot to say it earlier -- thanks for fixing the docs!

**Pull request merged**